src/query/api/v1/handler/placement/add.go

```diff
@@ -115,7 +131,17 @@ func (h *AddHandler) Add(
 		return nil, err
 	}
 
-	service, algo, err := ServiceWithAlgo(h.client, httpReq.Header)
+	serviceOpts := NewServiceOptionsFromHeaders(serviceName, httpReq.Header)
+	if serviceName == M3AggServiceName {
```

Comment:
nit: Maybe use a switch here so when we add further specialization later it's straightforward where to make the change (and we don't use an if/else if)?